apps/qa/src/template/template.py

```diff
@@ -0,0 +1,110 @@
+import streamlit as st
+from src.components import sidebar
+from src.components.build_outputs import (
+    load_build_outputs,
+    generate_geo_data,
+    generate_maps,
+    show_build_output,
+    show_map,
+)
+
+
+def template():
+    st.title("QA of Template DB")
+    product_key = sidebar.data_selection("db-template")
+
+    csv_files = [
+        "templatedb.csv",
+        "source_data_versions.csv",
+    ]
+
+    st.markdown(
+        body="""
+        #### [Product wiki](https://github.com/NYCPlanning/data-engineering/wiki/Product:-TDB)
+        
+        This QA page highlights key scenarios that can indicate potential data issues in a build.
+        """
+    )
+
+    if not product_key:
+        st.warning("No build version selected")
+        return
+
+    if "build_outputs" not in st.session_state:
+        st.session_state["build_outputs"] = {}
```

Comment:
I have a little bit of a gut response to avoid using session_state, though it does simplify code a good amount (and let you render the whole page while things are loading without crazy code). One note is that to generalize, we'll need to specify which product page is being viewed. And we might have to think about memory concerns if we're caching a bunch of big gdfs and maps for multiple products. But that's a problem that we deal with when it comes up 🙂

Reply:
yea I've always tried to avoid using session_state as much as I can, but the streamlit docs kept pushing me toward using it for this page's approach

I think it's also why I kept it's use in this new TDB page and not in any components meant to be shared by pages to not impose its use